src/framework/components/anim/system.js

```diff
@@ -89,6 +89,22 @@ class AnimComponentSystem extends ComponentSystem {
     }
 
     cloneComponent(entity, clone) {
+        let masks;
+        // If the component animaites from the components entity, any layer mask hierarchy should be updated from the old entity to the cloned entity.
+        if (!entity.anim.rootBone || entity.anim.rootBone === entity) {
+            masks = {};
+            entity.anim.layers.forEach((layer, i) => {
+                if (layer.mask) {
+                    const mask = {};
+                    Object.keys(layer.mask).forEach((path) => {
+                        // The base of all mask paths should be mapped from the previous entity to the cloned entity
+                        const clonePath = path.replace(path.split('/')[0], clone.name);
```

Comment:
could you also improve this line to make sure it only renames the first segment of the path (that's what needs to happen right?) .. for example replacing `root` by `clone` in `/root/rooter/something' would not give us expected result.